src/test/java/com/jcabi/s3/AwsBucketITCase.java

```diff
@@ -147,4 +150,17 @@ public void listsInPrefixedBucketWithouCollisions() throws Exception {
         }
     }
 
+    /**
+     * AwsBucket can check if the bucket exists.
+     * @throws IOException If fails
+     */
+    @Test
+    public void checkBucketExistence() throws IOException {
+        final Bucket bucket = this.rule.bucket();
+        Assert.assertTrue(bucket.exists());
```

Comment:
here is what may happen: 1) we create a real S3 bucket with `this.rule.bucket()`; 2) our method `exists()` fails because of some reason; 3) the entire build fails; 4) I fix the method `exists()`; 5) I run the build again; 6) the bucket stays in S3 forever. It's dirty. The temporary resource (S3 bucket) is not destroyed correctly.